pkgs/development/compilers/llvm/4/llvm.nix

```diff
@@ -110,6 +138,17 @@ in stdenv.mkDerivation (rec {
   ++ stdenv.lib.optionals (isDarwin) [
     "-DLLVM_ENABLE_LIBCXX=ON"
     "-DCAN_TARGET_i386=false"
+  ]
+  ++ stdenv.lib.optionals crossCompiling [
+    "-DCMAKE_CROSSCOMPILING=True"
```

Comment:
The target info is needed because we build compiler-rt with LLVM, right?

Reply:
I'm not sure, perhaps @bgamari can comment, I think he originally added this (or similar).

[LLVM cmake docs](https://llvm.org/docs/CMake.html) say this variable controls the what architecture is considered "native" for JIT.  So... should this be based on hostPlatform instead? Err....